### ISSUE TITLE
CI: LongRareTerms failing with NPE

### ISSUE THREAD

**Issue by costin** [elastic/elasticsearch#65301]
**Build scan**:
https://gradle-enterprise.elastic.co/s/2ilieh4lpxote

**Repro line**:

/gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.terms.LongRareTermsTests.testReduceRandom" -Dtests.seed=4439B24ABA377C7F -Dtests.security.manager=true -Dtests.locale=cs -Dtests.timezone=Australia/Lindeman -Druntime.java=8

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.terms.LongRareTermsTests.testReduceRandom" -Dtests.seed=4439B24ABA377C7F -Dtests.security.manager=true -Dtests.locale=cs -Dtests.timezone=Australia/Lindeman -Druntime.java=8

**Reproduces locally?**:
Yes.

**Applicable branches**:
7.10

**Failure history**:
https://build-stats.elastic.co/goto/0fae824715006863554c738c6df7f5ab

Looks to have started around 2 days ago, on the 18th.

**Failure excerpt**:

```
      at __randomizedtesting.SeedInfo.seed([4439B24ABA377C7F:84041E5B9C6AE529]:0)
        at org.elasticsearch.search.aggregations.bucket.terms.InternalMappedRareTerms.writeTermTypeInfoTo(InternalMappedRareTerms.java:84)
        at org.elasticsearch.search.aggregations.bucket.terms.InternalRareTerms.doWriteTo(InternalRareTerms.java:143)
        at org.elasticsearch.search.aggregations.InternalAggregation.writeTo(InternalAggregation.java:202)
        at org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteable(StreamOutput.java:1115)
        at org.elasticsearch.test.ESTestCase.lambda$copyNamedWriteable$15(ESTestCase.java:1310)
        at org.elasticsearch.test.ESTestCase.copyInstance(ESTestCase.java:1318)
        at org.elasticsearch.test.ESTestCase.copyNamedWriteable(ESTestCase.java:1309)
        at org.elasticsearch.test.ESTestCase.copyNamedWriteable(ESTestCase.java:1300)
        at org.elasticsearch.test.InternalAggregationTestCase.testReduceRandom(InternalAggregationTestCase.java:394)
```

**Comment by nik9000** [comment]
This is me. I'll have a look in a few hours. Feel free to mute if it fails
a bunch.

**Comment by nik9000** [comment]
This one looks to be caused by #64454. I've pinged @swallez to see if he'd like to take it.

**Comment by swallez** [comment]
@nik9000 looking at it!

**Comment by swallez** [comment]
@nik9000 

The tests fails only with this seed. Debugging to understand why `filter` is `null` and causes this exception, I found that in `InternalMappedRareTerms` the `filter` variable ends up being `null` if all aggregations have `isMapped() == false`, which turns out to be the case with this seed.

https://github.com/elastic/elasticsearch/blob/f7e840b012f6778cadb4d9e52fdbe44374c21ac5/server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalMappedRareTerms.java#L96-L98

I don't know what should be the correct filter value in that case, so can you take back this issue?

**Comment by nik9000** [comment]
Closed by 956ae93fdb9db47c4a55bb201eddae0f05bbd2d4 and 56605e4d9a7ca8423a85d9333ce12a4723c35d9e, both backports.
